dart/lib/src/hub.dart

```diff
@@ -65,12 +66,18 @@ class Hub {
       );
     } else {
       final item = _peek();
+      var scope = item.scope;
+
+      if (withScope != null) {
+        scope = item.scope.clone();
```

Comment:
```suggestion
        scope = scope.clone();
```

could be `scope` directly as you already assigned up there, right? if this is possible, fix the other cases too

Reply:
Good catch